nuxeo-features/nuxeo-automation/nuxeo-automation-features/src/main/java/org/nuxeo/ecm/automation/core/operations/notification/MailTemplateHelper.java

```diff
@@ -20,40 +20,46 @@
 
 import java.net.URL;
 
-import org.nuxeo.ecm.core.api.DocumentLocation;
 import org.nuxeo.ecm.core.api.DocumentModel;
-import org.nuxeo.ecm.core.api.impl.DocumentLocationImpl;
 import org.nuxeo.ecm.platform.ec.notification.service.NotificationService;
 import org.nuxeo.ecm.platform.ec.notification.service.NotificationServiceHelper;
 import org.nuxeo.ecm.platform.url.DocumentViewImpl;
 import org.nuxeo.ecm.platform.url.api.DocumentView;
 import org.nuxeo.ecm.platform.url.api.DocumentViewCodecManager;
+import org.nuxeo.ecm.platform.url.codec.api.DocumentViewCodec;
 import org.nuxeo.runtime.api.Framework;
 
 /**
  * @author <a href="mailto:bs@nuxeo.com">Bogdan Stefanescu</a>
  */
 public class MailTemplateHelper {
 
+    public static final String NOTIFICATION_DOCUMENT_ID_CODEC_NAME = "notificationDocId";
+
+    public static final String JSF_NOTIFICATION_DOCUMENT_ID_CODEC_PREFIX = "nxdoc";
+
     private MailTemplateHelper() {
     }
 
     public static String getDocumentUrl(DocumentModel doc, String viewId) {
-        if (viewId == null) {
-            viewId = "view_documents";
-        }
-        DocumentLocation docLoc = new DocumentLocationImpl(doc);
-        DocumentView docView = new DocumentViewImpl(docLoc);
-        docView.setViewId(viewId);
-        DocumentViewCodecManager codecMgr = Framework.getService(DocumentViewCodecManager.class);
-        NotificationService notifMgr = NotificationServiceHelper.getNotificationService();
-        if (codecMgr == null) {
-            throw new RuntimeException("Service 'DocumentViewCodecManager' not available");
-        }
-        if (notifMgr == null) {
-            throw new RuntimeException("Service 'NotificationService' not available");
+        NotificationService notificationService = NotificationServiceHelper.getNotificationService();
+        DocumentViewCodecManager codecService = Framework.getService(DocumentViewCodecManager.class);
+        DocumentViewCodec codec = codecService.getCodec(NOTIFICATION_DOCUMENT_ID_CODEC_NAME);
+        boolean isNotificationCodec = codec != null;
+
+        String result = "";
```

Comment:
Might want to throw an exception instead of returning "" or at least log something if no notification codec has been contributed?